ui/src/components/Tasks/AddSample.js

```diff
@@ -1,138 +1,138 @@
-    this.props.addToQueue(sampleData);
-    this.props.hide();
-  }
-
-  addAndEnqueue() {
-    this.props.handleSubmit(this._addAndEnqueue)();
-  }
+  useEffect(() => {
+    if (show) {
+      // Timeout required when creating new sample from "Samples" page
+      setTimeout(() => setFocus('sampleName'), 0);
+    }
+  }, [setFocus, show]);
 
-  _addAndMount(params) {
-    const sampleData = this.getDefaultSampleData(params);
-    this.props.addAndMount(sampleData);
-    this.props.hide();
-  }
+  function addAndMount(params) {
+    const sampleData = getSampleData(params);
+    addSamplesToList([sampleData]);
+    addSampleAndMount(sampleData);
+    hide();
 
-  addAndMount() {
-    this.props.handleSubmit(this._addAndMount)();
+    if (pathname === '/' || pathname === '/datacollection') {
+      // Switch to mounted sample tab
+      showList('current');
+    }
   }
 
-  handleKeyPress(target) {
-    if (target.charCode === 13) {
-      this.addAndMount();
-    }
+  function addAndQueue(params) {
+    const sampleData = getSampleData(params);
+    addSamplesToList([sampleData]);
+    addSamplesToQueue([sampleData]);
+    hide();
   }
 
-  render() {
-    return (
-      <Modal
-        // fullscreen
-        show={this.props.show}
-        onHide={this.handleCancel}
-        // style={{ width: '60%', height: '20%'}}
-      >
+  return (
+    <Modal show={show} onHide={hide} data-default-styles>
+      <Form noValidate onSubmit={handleSubmit(addAndMount)}>
         <Modal.Header closeButton>
-          <Modal.Title>Add Sample Manually</Modal.Title>
+          <Modal.Title>New Sample</Modal.Title>
         </Modal.Header>
         <Modal.Body>
-          <Form>
-            <FieldsRow>
-              <InputField
-                propName="sampleName"
-                autoFocus
-                label="Sample Name"
-                ref={(ref) => {
-                  this.sampleName = ref;
-                }}
-                col1="4"
-                col2="6"
+          <Form.Group as={Row} className="mb-3" controlId="sampleName">
+            <Col sm={4}>
+              <Form.Label column>Sample name</Form.Label>
+            </Col>
+            <Col sm={8}>
+              <Form.Control
+                type="text"
+                {...register('sampleName', {
+                  required: REQUIRED_MSG,
+                  pattern: { value: PATTERN, message: PATTERN_MSG },
+                })}
+                isValid={isSubmitted && !errors.sampleName}
+                isInvalid={isSubmitted && !!errors.sampleName}
               />
-              <InputField
-                propName="proteinAcronym"
-                label="Protein Acronym"
-                ref={(ref) => {
-                  this.proteinAcronym = ref;
-                }}
-                col1="4"
-                col2="6"
-                onKeyPress={this.handleKeyPress}
+              <Form.Control.Feedback type="invalid">
+                {errors.sampleName?.message}
```

Comment:
Bootstrap hides this error feedback until the field that precedes it is marked as invalid, so we don't need to render it conditionally.